File: LeetCode/008 String to Integer (atoi)/Solution.java
public class Solution {
    public int myAtoi(String str) {
        int count = 0;
        double rslt = 0;
        int sign = 1;
        char[] nosArray = str.toCharArray();

        for(char c: nosArray){
            count++;

            if (c>='0' && c<='9'){
                rslt = rslt*10 + (c-'0');
                //count--;
            }
            else if (c == '-' && count == 1){
                sign = -1;
            }
            else if (c == '+' && count == 1){
                sign = 1;
            }
            else if (c == ' ' && count == 1){
                count--;
            }
            else
                break;
        }

        if (rslt > Integer.MAX_VALUE){
            if(sign == 1)
                return Integer.MAX_VALUE;
            else
                return Integer.MIN_VALUE;

        }
        else
            return (int)rslt*sign;
    }
}

File: LeetCode/008 String to Integer (atoi)/Solution.py
class Solution:
    def myAtoi(self, st):
        """
        :type str: str
        :rtype: int
        """
        INT_MAX =  2147483647
        INT_MIN = -2147483648
        result = 0

        if not st:
            return result

        i = 0
        sign = 1

        while i < len(st) and st[i].isspace():
            i +=1

        if st[i] == "+":
            i +=1
        elif st[i] == "-":
            i +=1
            sign = -1

        while i < len(st) and '0' <= st[i] <= '9':
            if result > (INT_MAX - int(st[i]))/10:
                return INT_MAX if sign > 0 else INT_MIN

            result = result * 10 + int(st[i])
            i +=1

        return result * sign



if __name__ == '__main__':
    print(Solution().myAtoi(""))
    print(Solution().myAtoi("-1"))
    print(Solution().myAtoi("2147483647"))
    print(Solution().myAtoi("2147483648"))
    print(Solution().myAtoi("-2147483648"))
    print(Solution().myAtoi("-2147483649"))
